tagger.go

```diff
@@ -86,254 +111,324 @@ func newLinodeClient() linodego.Client {
 	return client
 }
 
-func compareTags(linodeObject interface{}, tagRules []TagRule) ([]string, []string, int) {
-	// returns a slice of new tags, a slice of old tags, and the object ID (in order)
-
-	var combinedNewTags, tags []string
-	var objectID int
-
-	var label string
-	switch object := linodeObject.(type) {
-	case linodego.Instance:
-		tags = object.Tags
-		label = object.Label
-		objectID = object.ID
-	case linodego.Volume:
-		tags = object.Tags
-		label = object.Label
-		objectID = object.ID
-	case linodego.NodeBalancer:
-		tags = object.Tags
-		label = *object.Label
-		objectID = object.ID
-	case linodego.Domain:
-		tags = object.Tags
-		label = object.Domain
-		objectID = object.ID
-	case linodego.LKECluster:
-		tags = object.Tags
-		label = object.Label
-		objectID = object.ID
-	}
-
-	sort.Strings(tags)
-
-	for _, rule := range tagRules {
-		validObject := regexp.MustCompile(rule.Regex)
-
-		if validObject.MatchString(label) {
-			var newTags []string
-
-			// check `absent` tags to remove unwanted tags
-			for _, tag := range tags {
-				if !slices.Contains(rule.Tags.Absent, tag) {
-					// if this tag is not on the `absent` list,
-					// we can persist it through to the new tag set
-					newTags = append(newTags, tag)
-				}
-			}
+// getTagDiff accepts 2 string slices (the current set of tags, and the desired
+// set of tags). It outputs a Diff object containing the changes
+func getTagDiff(have, want []string) Diff {
+	var d Diff
+
+	sort.Strings(have)
+	sort.Strings(want)
+	if !slices.Equal(have, want) {
+		// order of have and object.have differs based on whether we're subtracting or contributing more have
+		d.Added = sliceDifference(want, have)
+		d.Removed = sliceDifference(have, want)
+	}
```

Comment:
```suggestion
// getTagDiff accepts 2 string slices (the current set of tags, and the desired
// set of tags). It outputs a Diff object containing the changes
func getTagDiff(have, want []string) Diff {
	var diff Diff

	sort.Strings(have)
	sort.Strings(want)
	if !slices.Equal(have, want) {
		// order of have and object.have differs based on whether we're subtracting or contributing more have
		diff.Added = sliceDifference(want, have)
		diff.Removed = sliceDifference(have, want)
	}
```
small improvement to readability